test/jdk/tools/launcher/modules/illegalaccess/IllegalAccessTest.java

```diff
@@ -380,12 +381,8 @@ public void testWithAddExportsInManifest() throws Exception {
 
         run(jarfile, "reflectPublicMemberNonExportedPackage", successNoWarning());
 
-        run(jarfile, "setAccessibleNonPublicMemberExportedPackage", successWithWarning());
-
-        // attempt two illegal accesses, one allowed by Add-Exports
-        run(jarfile, "reflectPublicMemberNonExportedPackage,"
-                + "setAccessibleNonPublicMemberExportedPackage",
```

Comment:
I see `setAccessibleNonPublicMemberExportedPackage` test case testing with `--add-exports` and `--add-opens` and `Add-Opens` in JAR file manifest but not testing with `Add-Exports`.

Is it worth to include the `Add-Exports` test case?

Other than this, looks good.

Reply:
Good point -- case added.

Reply:
Looks good.

Reply:
testWithAddExportsInManifest create an executable JAR with "Add-Exports: java.base/sun.security.x509" in the manifest. It runs it twice, once without any options, the second with --illegal-access=permit, and checks there are no warnings in both cases. To test attempted deep reflection here would need the equivalent of setAccessibleNonPublicMemberNonExportedPackage that tries to access some privates in sun.security.x509. It's okay to use setAccessibleNonPublicMemberNonExportedPackage to test that privates in sun.nio.ch aren't accessible but it's not connected to the Add-Exports attribute in the JAR file.

Reply:
Thanks for pointing that out. I fixed it by arranging for the `Add-Exports` test to export both `sun.security.x509` and `sun.nio.ch`.